src/main/groovy/org/jetbrains/intellij/tasks/RunPluginVerifierTask.groovy

```diff
@@ -0,0 +1,232 @@
+    private String pluginsToCheckFile = ""
+    private String subsystemsToCheck = ""
+
+    @Input
+    List<String> getIdes() {
+        return ides
+    }
+
+    void setIdes(List<String> ides) {
+        this.ides = ides
+    }
+
+    @Input
+    String getVerifierVersion() {
+        return verifierVersion
+    }
+
+    void setVerifierVersion(String verifierVersion) {
+        this.verifierVersion = verifierVersion
+    }
+
+    @Input
+    String getVerificationReportsDir() {
+        return verificationReportsDir
+    }
+
+    void setVerificationReportsDir(String verificationReportsDir) {
+        this.verificationReportsDir = verificationReportsDir
+    }
+
+    @Input
+    String getRuntimeDir() {
+        return runtimeDir
+    }
+
+    void setRuntimeDir(String runtimeDir) {
+        this.runtimeDir = runtimeDir
+    }
+
+    @Input
+    String getExternalPrefixes() {
+        return externalPrefixes
+    }
+
+    void setExternalPrefixes(String externalPrefixes) {
+        this.externalPrefixes = externalPrefixes
+    }
+
+    @Input
+    String getPluginsToCheckAllBuilds() {
+        return pluginsToCheckAllBuilds
+    }
+
+    void setPluginsToCheckAllBuilds(String pluginsToCheckAllBuilds) {
+        this.pluginsToCheckAllBuilds = pluginsToCheckAllBuilds
+    }
+
+    @Input
+    String getPluginsToCheckLastBuilds() {
+        return pluginsToCheckLastBuilds
+    }
+
+    void setPluginsToCheckLastBuilds(String pluginsToCheckLastBuilds) {
+        this.pluginsToCheckLastBuilds = pluginsToCheckLastBuilds
+    }
+
+    @Input
+    String getTeamCity() {
+        return teamCity
+    }
+
+    void setTeamCity(String teamCity) {
+        this.teamCity = teamCity
+    }
+
+    @Input
+    String getTcGrouping() {
+        return tcGrouping
+    }
+
+    void setTcGrouping(String tcGrouping) {
+        this.tcGrouping = tcGrouping
+    }
+
+    @Input
+    String getExcludedPluginsFile() {
+        return excludedPluginsFile
+    }
+
+    void setExcludedPluginsFile(String excludedPluginsFile) {
+        this.excludedPluginsFile = excludedPluginsFile
+    }
+
+    @Input
+    String getDumpBrokenPluginList() {
+        return dumpBrokenPluginList
+    }
+
+    void setDumpBrokenPluginList(String dumpBrokenPluginList) {
+        this.dumpBrokenPluginList = dumpBrokenPluginList
+    }
+
+    @Input
+    String getPluginsToCheckFile() {
+        return pluginsToCheckFile
+    }
+
+    void setPluginsToCheckFile(String pluginsToCheckFile) {
+        this.pluginsToCheckFile = pluginsToCheckFile
+    }
+
+    @Input
+    String getSubsystemsToCheck() {
+        return subsystemsToCheck
+    }
+
+    void setSubsystemsToCheck(String subsystemsToCheck) {
+        this.subsystemsToCheck = subsystemsToCheck
+    }
+
+    @TaskAction
+    void runPluginVerifier() {
+        def extension = project.extensions.findByType(IntelliJPluginExtension)
+        def pluginFileName = "${extension.pluginName}-${project.version}"
```

Comment:
@zolotov Can you assist with that? WDYT about the current approach and Yann's concerns?

Reply:
agree with Yann. You should accept plugin file outside. It will fix the problem with the name + it makes your task more incremental (no check for unmodified plugin) + it will be able to run verification on several plugins. See PublishTask for reference